step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/BodyExecutionCallback.java

```diff
@@ -94,4 +94,38 @@ public void onFailure(StepContext context, Throwable t) {
 
     private static final long serialVersionUID = 1L;
 
+    /**
+     * A convenience subclass for the common case that the step expects to run its block just once and return the same value (or throw the same error).
+     */
+    public static abstract class TailCall extends BodyExecutionCallback {
```

Comment:
Fair enough. Maybe add a comment to that effect (or a link to that Wikipedia page) for dummies like me?

NIT: Still seems like impls of this are all callbacks (`TailCallCallback`).

:bee: